src/value.rs

```diff
@@ -32,6 +32,20 @@ impl<'a, W: Write> ToMemcacheValue<W> for &'a [u8] {
     }
 }
 
+impl<'a, W: Write> ToMemcacheValue<W> for &'a String {
```

Comment:
Looks like implement the trait for `&String` have no sense for we had implemented the trait for `&str`?

But looks like this may be convenient in some case like we do in the test, so I think we should merge it. :+1: